jib-cli/src/test/java/com/google/cloud/tools/jib/cli/cli2/CredentialsTest.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.cli.cli2;
+
+import com.google.cloud.tools.jib.api.Credential;
+import com.google.cloud.tools.jib.plugins.common.DefaultCredentialRetrievers;
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import org.apache.commons.lang3.ArrayUtils;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+import picocli.CommandLine;
+
+import java.io.FileNotFoundException;
+
+import static com.google.common.truth.Truth.assertThat;
+
+@RunWith(JUnitParamsRunner.class)
+public class CredentialsTest {
+
+  private static final String[] DEFAULT_ARGS = {"--target=ignored"};
+  @Rule public final MockitoRule mockitoJUnit = MockitoJUnit.rule();
+  @Mock private DefaultCredentialRetrievers defaultCredentialRetrievers;
+
+  private Object paramsToNone() {
+    return new Object[]{
+        new String[]{"--from-credential-helper=ignored"},
+        new String[]{"--from-username=ignored", "--from-password=ignored"},
+    };
+  }
+
+  @Test
+  @Parameters(method = "paramsToNone")
```

Comment:
optional: Not sure how this would look but how about specifying the params directly in the tests instead of calling a method?

Reply:
It does appear to work for array types.

Reply:
Probably @loosebazooka meant "doesn't."